### PR TITLE
footprint shrink a bit

### PR DESCRIPTION
**Pull request by kimstik**
footprint is 727 bytes (instead of 1389 from last iteration)
get rid of useless runtime 32bit divisions
FIXME: makefile was screwed for non-bootloader apps
TODO: lets pack to 512?

### CODE REVIEW

**Comment by wkaster** [comment]
Very Nice! I'll test this branch before merge.   Thinking about create a "standalone" option in the makefile to supress the lib directory if needed.

**Comment by wkaster** [comment]
Bug: Not receiving the CMD_ACK after software reset request (CMD_EOT), dont know why.
Aparently handshake is ok, flash program is ok.
Tested with 2 SDCC versions:
3.91 footprint 754
4.1 footprint 723

File: N76_bootloader/bootloader/main.c [hunk 1]
```diff
@@ -28,11 +28,39 @@
...
+#ifdef FOSC_166000
+	RH3    = HIBYTE(65536 - (1037500/u32Baudrate)); 		/*16.6 MHz */
+	RL3    = LOBYTE(65536 - (1037500/u32Baudrate)); 		/*16.6 MHz */
+#endif
+	set_TR3;         //Trigger Timer3
+	set_TI;					 //For printf function must setting TI = 1
+}
+
+static void tx(char c) {
+	while(TI==0);
```
Comment (wkaster) [inline N76_bootloader/bootloader/main.c]: This line should be the last in this function.  This is causing the software reset before ACK is send,

**Comment by wkaster** [comment]
Bug found:  tx function not sending the ACK before software reset.  "while(TI==0);" should go to the last line.

**Comment by kimstik** [comment]
I believe that "while(TI==0);" placed correctly
This way is more correct and secure. Plus it gives better performance.

**Comment by kimstik** [comment]
May you test last one before merging?

**Comment by wkaster** [comment]
> I believe that "while(TI==0);" placed correctly This way is more correct and secure. Plus it gives better performance.

Yes, your way is better.

**Comment by wkaster** [comment]
> May you test last one before merging?

Of course, Do you mind if I add your name to the source header as co-author?

**Comment by wkaster** [comment]
Could not make those last 2 commits to work.  They simply don't boot. (SDCC 4.1)

**Comment by kimstik** [comment]
> Could not make those last 2 commits to work. They simply don't boot. (SDCC 4.1)

Really? No any answer for CMD_SOH ?
I see that erase screwed in last commit.
is 4ff1bf4 commit works?

**Comment by wkaster** [comment]
Last working version: [7bfc15f] WHITOUT the modified peep file.  I'll do some more testing as soon as possible.

**Review by wkaster — COMMENTED** [review]
This one is working fine!

**Comment by kimstik** [comment]
> Of course, Do you mind if I add your name to the source header as co-author?

I hesitate to change the header.
You can add me as a contributor.

**Comment by wkaster** [comment]
> > Of course, Do you mind if I add your name to the source header as co-author?
> 
> I hesitate to change the header. You can add me as a contributor.

In my mind, with the work you have done so far, this bootloader is much yours as it is mine.   Please, feel free to change the header as you see fit.

**Pull request merged**